From fb8d5396745c1f52dcc0883a6c62e0c5a11d2262 Mon Sep 17 00:00:00 2001
From: xtex <xtex@astrafall.org>
Date: Sun, 8 Mar 2026 08:53:48 +0800
Subject: [PATCH 127/133] std.os.linux: add LoongArch HWCAP bits
X-Developer-Signature: v=1; a=openpgp-sha256; l=1160; i=xtex@astrafall.org;
 h=from:subject; bh=8r7u3KtrYE9JHNxQsfqii7eouYMrJC4HefpQEF24RPw=;
 b=owGbwMvMwCW2U4Ij7wZL9ETG02pJDJn78gRe+4X8ds6WbOW6uvIc8xE97Tn/g847ar+RCAi/b
 bj/06mIjoksDGJcDJZiiixFhg3erDrp/KLLymVh5rAygQyRFmlgYGBgYGHgy03MKzXSMdIz1TbU
 MzTUMdIxYuDiFICp3r+G4Z8S27Mn4dUxzqF/i/+yhkx0ezTLvaf4h/KOxLAPflpyOz4zMhy9l3X
 7+xT3Ks5K0/Pux0ptlzV+d1JsmVFg/z7fJoZdkhUA
X-Developer-Key: i=xtex@astrafall.org; a=openpgp;
 fpr=7231804B052C670F15A6771DB918086ED8045B91

See also arch/loongarch/include/uapi/asm/hwcap.h in Linux kernel.

SC.Q was added in kernel v7.0.

Change-Id: Ib4ddd451ab16d52b2578f8b97c9c3365286719cd
Signed-off-by: xtex <xtex@astrafall.org>
---
 lib/std/os/linux/loongarch64.zig | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/lib/std/os/linux/loongarch64.zig b/lib/std/os/linux/loongarch64.zig
index 2399353fa7f2..e2218e32f94a 100644
--- a/lib/std/os/linux/loongarch64.zig
+++ b/lib/std/os/linux/loongarch64.zig
@@ -246,3 +246,22 @@ pub const Elf_Symndx = u32;
 
 /// TODO
 pub const getcontext = {};
+
+pub const HWCAP = struct {
+    pub const CPUCFG = 1 << 0;
+    pub const LAM = 1 << 1;
+    pub const UAL = 1 << 2;
+    pub const FPU = 1 << 3;
+    pub const LSX = 1 << 4;
+    pub const LASX = 1 << 5;
+    pub const CRC32 = 1 << 6;
+    pub const COMPLEX = 1 << 7;
+    pub const CRYPTO = 1 << 8;
+    pub const LVZ = 1 << 9;
+    pub const LBT_X86 = 1 << 10;
+    pub const LBT_ARM = 1 << 11;
+    pub const LBT_MIPS = 1 << 12;
+    pub const PTW = 1 << 13;
+    pub const LSPW = 1 << 14;
+    pub const SCQ = 1 << 15;
+};
-- 
2.52.0

